crates/networking/p2p/discv4/server.rs

```diff
@@ -2,141 +2,330 @@ use std::{
+
+                if is_msg_expired(ping_message.expiration) {
+                    trace!("Ping expired, skipped");
+                    return;
+                }
+
+                let node = Node::new(
+                    unmap_ipv4in6_address(from.ip()),
+                    from.port(),
+                    ping_message.from.tcp_port,
+                    sender_public_key,
+                );
+
+                let _ = self.handle_ping(hash, node).await.inspect_err(|e| {
+                    error!(sent = "Ping", to = %format!("{sender_public_key:#x}"), err = ?e);
+                });
+            }
+            Message::Pong(pong_message) => {
+                trace!(received = "Pong", msg = ?pong_message, from = %format!("{:#x}", sender_public_key));
+
+                let node_id = node_id(&sender_public_key);
+
+                self.handle_pong(pong_message, node_id).await;
+            }
+            Message::FindNode(find_node_message) => {
+                trace!(received = "FindNode", msg = ?find_node_message, from = %format!("{:#x}", sender_public_key));
+
+                if is_msg_expired(find_node_message.expiration) {
+                    trace!("FindNode expired, skipped");
+                    return;
+                }
+
+                self.handle_find_node(sender_public_key, from).await;
+            }
+            Message::Neighbors(neighbors_message) => {
+                trace!(received = "Neighbors", msg = ?neighbors_message, from = %format!("{sender_public_key:#x}"));
+
+                if is_msg_expired(neighbors_message.expiration) {
+                    trace!("Neighbors expired, skipping");
+                    return;
+                }
+
+                self.handle_neighbors(neighbors_message).await;
+            }
+            Message::ENRRequest(enrrequest_message) => {
+                trace!(received = "ENRRequest", msg = ?enrrequest_message, from = %format!("{sender_public_key:#x}"));
+
+                if is_msg_expired(enrrequest_message.expiration) {
+                    trace!("ENRRequest expired, skipping");
+                    return;
+                }
+
+                self.handle_enr_request(sender_public_key, from, hash).await;
+            }
+            Message::ENRResponse(enrresponse_message) => {
+                /*
+                    - Look up in kademlia the peer associated with this message
+                    - Check that the request hash sent matches the one we sent previously (this requires setting it on enrrequest)
+                    - Check that the seq number matches the one we have in our table (this requires setting it).
+                    - Check valid signature
+                    - Take the `eth` part of the record. If it's None, this peer is garbage; if it's set
+                */
+                trace!(received = "ENRResponse", msg = ?enrresponse_message, from = %format!("{sender_public_key:#x}"));
```

Comment:
this is a TODO?

Reply:
Yes, it seems so, adding a "TODO" on top
(It's just moved code, the intention of this refactor is not to add behavior, just clean up and implement proper spawned actors)